phpBB/posting.php

```diff
@@ -87,24 +87,25 @@
 		$post_id = $request->variable('p', 0);
 		if ($post_id)
 		{
-			$topic_forum = array();
+			$topic_forum = [];
 
 			$sql = 'SELECT t.topic_id, t.forum_id
 				FROM ' . TOPICS_TABLE . ' t, ' . POSTS_TABLE . ' p
 				WHERE p.post_id = ' . $post_id . '
 				AND t.topic_id = p.topic_id';
 			$result = $db->sql_query($sql);
-			$topic_forum = $db->sql_fetchrow();
-			$topic_id = (int) $topic_forum['topic_id'];
-			$forum_id = (int) $topic_forum['forum_id'];
+			$topic_forum = $db->sql_fetchrow($result);
 			$db->sql_freeresult($result);
 		}
 
-		if (!$post_id || !$topic_id || !$forum_id)
+		if (!$post_id || !$topic_forum)
```

Comment:
May I suggest what I wrote to solve this ticket, which is slightly different?

Reply:
Sure, what code do you have in mind?

Reply:
Well, I did delete it. Here is what I wrote now and briefly tested.

```php
	case 'edit':
	case 'delete':
	case 'quote':
	case 'soft_delete':
		$post_id = $request->variable('p', 0);

		$sql = 'SELECT post_id
			FROM ' . POSTS_TABLE . '
			WHERE post_id = ' . $post_id;
		$result = $db->sql_query_limit($sql, 1);
		$row = $db->sql_fetchrow($result);
		$db->sql_freeresult($result);

		$post_id = $row['post_id'] ?? 0;

		if ($post_id)
		{
			$topic_forum = [];

			$sql = 'SELECT t.topic_id, t.forum_id
				FROM ' . TOPICS_TABLE . ' t, ' . POSTS_TABLE . ' p
				WHERE p.post_id = ' . $post_id . '
				AND t.topic_id = p.topic_id';
			$result = $db->sql_query($sql);
			$topic_forum = $db->sql_fetchrow($result);
			$db->sql_freeresult($result);

			$topic_id = (int) $topic_forum['topic_id'];
			$forum_id = (int) $topic_forum['forum_id'];
		}
		else
		{
			$user->setup('posting');
			trigger_error('NO_POST');
		}
	break;
}
```

Reply:
Well, that additional query actually duplicates the current query (the only difference is the current one joins topics table, but this doesn't change the result).
I'd think about adding a limit of 1 to the query, not sure if it brings any benefits though.

Reply:
There is no need to go ahead if the post does not exists and a simple query which asserts it does or not simpliefies the workflow. The same method should also be used above in the same swittch/cases for the forums (IMHO).

Reply:
On a side note, is this is a regression due to the same PR which took care of removing the forum queries from links?

Reply:
Sorry, I'm still on the point those queries do the same thing :)

> is this is a regression due to the same PR which took care of removing the forum queries from links?

Yes, the whole code was added here
https://github.com/phpbb/phpbb/commit/adacf21edb0bdd3e8aaa649ca037586a68ccf5e2#diff-2bd852c3b9a70bfaabb479ef5b67b657056c1e74de09e433e52a644092ec28c9

Reply:
Two different points of view, indeed. :)

Reply:
Again, ther is no need to run a query which joins two tables if the post ID does not exist to begin with, you can not rely on the request var to assert it.

Reply:
Post existence is checked with that query. There's no need in additional one. There's no relying on single request value.